app/screens/more_dms/more_dms.tsx

```diff
@@ -0,0 +1,409 @@
+        let success;
+        if (idsToUse.length === 0) {
+            success = false;
+        } else if (idsToUse.length > 1) {
+            success = await createGroupChannel(idsToUse);
+        } else {
+            success = await createDirectChannel(idsToUse[0]);
+        }
+
+        if (success) {
+            requestAnimationFrame(() => {
+                close();
+            });
+        } else {
+            setStartingConversation(false);
+        }
+    }, [startingConversation, selectedIds, createGroupChannel, createDirectChannel]);
+
+    const handleSelectProfile = useCallback((user: UserProfile) => {
+        if (selectedIds[user.id]) {
+            handleRemoveProfile(user.id);
+            return;
+        }
+
+        if (user.id === currentUserId) {
+            const selectedId = {
+                [currentUserId]: true,
+            };
+
+            startConversation(selectedId);
+        } else {
+            const wasSelected = selectedIds[user.id];
+
+            if (!wasSelected && selectedCount >= General.MAX_USERS_IN_GM - 1) {
+                return;
+            }
+
+            const newSelectedIds = Object.assign({}, selectedIds);
+            if (!wasSelected) {
+                newSelectedIds[user.id] = user;
+            }
+
+            setSelectedIds(newSelectedIds);
+
+            clearSearch();
+        }
+    }, [selectedIds, currentUserId, handleRemoveProfile, startConversation, clearSearch]);
+
+    const searchUsers = useCallback(async (searchTerm: string) => {
+        const lowerCasedTerm = searchTerm.toLowerCase();
+        setLoading(true);
+        let results;
+
+        if (restrictDirectMessage) {
+            results = await searchProfiles(serverUrl, lowerCasedTerm);
+        } else {
+            results = await searchProfiles(serverUrl, lowerCasedTerm, {team_id: currentTeamId});
+        }
+
+        let data: UserProfile[] = [];
+        if (results.data) {
+            data = results.data;
+        }
+
+        setSearchResults(data);
+        setLoading(false);
+    }, [restrictDirectMessage, serverUrl, currentTeamId]);
+
+    const onSearch = useCallback((text: string) => {
+        if (text) {
+            setTerm(text);
+            if (searchTimeoutId.current) {
+                clearTimeout(searchTimeoutId.current);
+            }
+
+            searchTimeoutId.current = setTimeout(() => {
+                searchUsers(text);
+            }, General.SEARCH_TIMEOUT_MILLISECONDS);
+        } else {
+            clearSearch();
+        }
+    }, [searchUsers, clearSearch]);
+
+    const updateNavigationButtons = useCallback(async (startEnabled: boolean) => {
+        const closeIcon = await CompassIcon.getImageSource('close', 24, theme.sidebarHeaderTextColor);
+        setButtons(componentId, {
+            leftButtons: [{
+                id: CLOSE_BUTTON,
+                icon: closeIcon,
+                testID: 'close.more_direct_messages.button',
+            }],
```

Comment:
the left button does not seem to change at any moment, perhaps best to open the modal and setting the left button there and only update the right button here? 0/5